src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/InterchangeHeader.java

```diff
@@ -0,0 +1,81 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A specialisation of a segment for the specific use case of an interchange header
+ * takes in specific values required to generate an interchange header
+ * example: UNB+UNOA:2+TES5+XX11+920113:1317+00000002'
+ */
+@Getter
+@Setter
+@NoArgsConstructor
+@RequiredArgsConstructor
+@AllArgsConstructor
+public class InterchangeHeader extends Segment {
+
+    public static final String KEY = "UNB";
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyMMdd:HHmm").withZone(TimestampService.UK_ZONE);
+    private static final long MAX_INTERCHANGE_SEQUENCE = 99_999_999L;
+    private static final String RECEP_ENDING = "+RECEP+++EDIFACT TRANSFER";
+
+    private @NonNull String sender;
+    private @NonNull String recipient;
+    private @NonNull Instant translationTime;
+    private Long sequenceNumber;
+
+    public static InterchangeHeader fromString(String edifactString) {
+        if (!edifactString.startsWith(InterchangeHeader.KEY)) {
+            throw new IllegalArgumentException("Can't create " + InterchangeHeader.class.getSimpleName() + " from " + edifactString);
+        }
+        String[] split = Split.byPlus(edifactString);
+
+        ZonedDateTime translationTime = ZonedDateTime.parse(split[4], DateTimeFormatter.ofPattern("yyMMdd:HHmm").withZone(TimestampService.UK_ZONE));
+        return new InterchangeHeader(split[2], split[3], translationTime.toInstant(), Long.valueOf(split[5]));
+    }
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        String timestamp = DATE_FORMAT.format(translationTime);
+        String formattedSequenceNumber = String.format("%08d", sequenceNumber);
+        return "UNOA:2" + "+" + sender + "+" + recipient + "+" + timestamp + "+" + formattedSequenceNumber;
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
```

Comment:
Just FYI, since we will be doing one-way translations, we might not need all those validation methods. Don't remove that now